policies/documentation-policy.md

```diff
@@ -0,0 +1,123 @@
+included in the `doc/man7` section.  This includes, but is not limited to:
+
+- descriptions of data structures and their usage
+- provider calls and call backs
+- algorithm descriptions and parameters
+- descriptions of architectural decisions
+
+## Chapter 4: Internal functions, structures, globals and macros
+
+Internal functions and macros are those defined in the `include/crypto`
+and `include/internal` directories.
+
+These should all be documented at the point of implementation.  A comment
+describing their purpose and the input and output arguments and return value
+for functions suffices.
+
+For _trivial_ items, where their operation is obvious from their
+implementation, the documentation requirement is not required.  The following
+are generally representative of trivial items, however it is quite
+possible for any of these to be non-trivial in specific instances and
+therefore require documentation:
+
+- OSSL_DISPATCH tables
+- upref functions
+- free functions
+- simple getter/setter functions
+- wrappers for other functions (a function that calls a more recent `_ex`
+  variant or a group of functions that call a common internal routine)
+
+For structures, each of the fields should be commented stating its
+purpose.  Again, a _trivial_ exception applies where the purpose is
+obvious.  Some representative examples:
+
+- `OSSL_LIB_CTX *ctx;` where there is only one library context referenced in
+  the structure.
+- `struct *next;` in a linked list implementation.
+- `CRYPTO_REF_COUNT refcnt;`
+- `CRYPTO_RWLOCK *lock;` where there is only one lock in the structure and
+  it locks the entire structure.
+
+## Chapter 5: Static functions and globals and local structures and macros
+
+These are functions, structures, globals and macros local to a specific
+C file or defined for a single directory as part of a local .h file.
+
+These should all be documented at the point of implementation.  Follow the
+rules and exceptions listed in Chapter 3.  In some cases slightly more
+leniency with respect to _trivial_ can be tolerated.
+
+## Chapter 6: Code comments
+
+Comments in code should explain what the algorithm is doing and why.
+It should not parrot the effect of each statement.  As the complexity of
+the code increases, the size and detail of comments should also increase.
+
+## Chapter 7: Assembly code
+
+Assembly code should include a good description of the algorithm and
+approach being used.  This should be followed by a performance comparison
+and then the assembly code itself.  The assembly code should be well
+commented, but it isn't necessary to comment every line.  A comment
+describing each block of code suffices.
+
+There are no _trivial_ exceptions for assembly code.
+
+## Chapter 8: Configure options
+
+New options added to the configuration scripts must be documents in the
+`INSTALL.md` file.
+
+## Chapter 9: Changes and news
+
+Non-trivial modification should be documented in the `CHANGES.md` file.
```

Comment:
"non-trivial" is quite a different bar to what we do at the moment. We generally reserve CHANGES.md updates to significant changes.